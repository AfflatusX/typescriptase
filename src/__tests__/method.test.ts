/**
 * This file is partially generated; only edit bespoke sections.
 *
 * SOURCE<<gen/tests.ts::methodTest>>
 * BESPOKE<<imports, beforeAll, afterAll, beforeEach, afterEach, methodInstance1, methodInstance2, methodInstance3, methodInstance4, methodInstance5, methodStatic1, methodStatic2, methodStatic3>>
 * SIGNED<<zjhk0tXB2WOsVyAGPte0RTRsWqaKNGpd6TvqM82IHKb7ZuPjr4yR+dAG2BhiqolRaNY3V0NBPFDTJelHqZ3w2g==>>
 */

/* BESPOKE START <<imports>> */
import {
  methodInstance1,
  methodInstance2,
  methodInstance3,
  methodInstance4,
  methodInstance5,
  methodStatic1,
  methodStatic2,
  methodStatic3,
} from "../../gen/examples";
/* BESPOKE END <<imports>> */

beforeAll(
  async (): Promise<void> => {
    /* BESPOKE START <<beforeAll>> */
    /* BESPOKE END <<beforeAll>> */
  },
);

afterAll(
  async (): Promise<void> => {
    /* BESPOKE START <<afterAll>> */
    /* BESPOKE END <<afterAll>> */
  },
);

beforeEach(
  async (): Promise<void> => {
    /* BESPOKE START <<beforeEach>> */
    /* BESPOKE END <<beforeEach>> */
  },
);

afterEach(
  async (): Promise<void> => {
    /* BESPOKE START <<afterEach>> */
    /* BESPOKE END <<afterEach>> */
  },
);

test(
  "methodInstance1",
  async (): Promise<void> => {
    /* BESPOKE START <<methodInstance1>> */
    expect(methodInstance1.print())
      .toMatchSnapshot();
    /* BESPOKE END <<methodInstance1>> */
  },
);

test(
  "methodInstance2",
  async (): Promise<void> => {
    /* BESPOKE START <<methodInstance2>> */
    expect(methodInstance2.print())
      .toMatchSnapshot();
    /* BESPOKE END <<methodInstance2>> */
  },
);

test(
  "methodInstance3",
  async (): Promise<void> => {
    /* BESPOKE START <<methodInstance3>> */
    expect(methodInstance3.print())
      .toMatchSnapshot();
    /* BESPOKE END <<methodInstance3>> */
  },
);

test(
  "methodInstance4",
  async (): Promise<void> => {
    /* BESPOKE START <<methodInstance4>> */
    expect(methodInstance4.print())
      .toMatchSnapshot();
    /* BESPOKE END <<methodInstance4>> */
  },
);

test(
  "methodInstance5",
  async (): Promise<void> => {
    /* BESPOKE START <<methodInstance5>> */
    expect(methodInstance5.print())
      .toMatchSnapshot();
    /* BESPOKE END <<methodInstance5>> */
  },
);

test(
  "methodStatic1",
  async (): Promise<void> => {
    /* BESPOKE START <<methodStatic1>> */
    expect(methodStatic1.print())
      .toMatchSnapshot();
    /* BESPOKE END <<methodStatic1>> */
  },
);

test(
  "methodStatic2",
  async (): Promise<void> => {
    /* BESPOKE START <<methodStatic2>> */
    expect(methodStatic2.print())
      .toMatchSnapshot();
    /* BESPOKE END <<methodStatic2>> */
  },
);

test(
  "methodStatic3",
  async (): Promise<void> => {
    /* BESPOKE START <<methodStatic3>> */
    expect(methodStatic3.print())
      .toMatchSnapshot();
    /* BESPOKE END <<methodStatic3>> */
  },
);
