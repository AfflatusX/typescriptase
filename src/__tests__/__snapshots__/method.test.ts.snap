// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`method1 1`] = `
"public async method1<T>(
  var1: string,
  var2: object = {},
): Promise<string[]> {
}
"
`;

exports[`method2 1`] = `
"protected static abstract method2<T, V>(
): void;
"
`;

exports[`method3 1`] = `
"private static async method3(
): void {
  /* BESPOKE START <<method3Bespoke>> */
  /* BESPOKE END <<method3Bespoke>> */
}
"
`;
